ts/src/profilers/heap-profiler.ts

```diff
@@ -15,41 +15,54 @@
  */
 
 import {perftools} from '../../../proto/profile';
+import {defaultConfig} from '../config';
+
 import {serializeHeapProfile} from './profile-serializer';
 
 const profiler = require('bindings')('sampling_heap_profiler');
 
-export class HeapProfiler {
-  private enabled = false;
+let enabled = false;
+let heapIntervalBytes = defaultConfig.heapIntervalBytes;
+let heapStackDepth = defaultConfig.heapMaxStackDepth;
 
-  /**
-   * @param intervalBytes - average bytes between samples.
-   * @param stackDepth - upper limit on number of frames in stack for sample.
-   */
-  constructor(private intervalBytes: number, private stackDepth: number) {
-    this.enable();
+/*
+ * Collects a heap profile when heapProfiler is enabled. Otherwise throws
+ * an error.
+ */
+export function profile(): perftools.profiles.IProfile {
+  if (!enabled) {
+    throw new Error('Heap profiler is not enabled.');
   }
+  const result = profiler.getAllocationProfile();
+  const startTimeNanos = Date.now() * 1000 * 1000;
+  return serializeHeapProfile(result, startTimeNanos, heapIntervalBytes);
+}
 
-  /**
-   * Collects a heap profile when heapProfiler is enabled. Otherwise throws
-   * an error.
-   */
-  profile(): perftools.profiles.IProfile {
-    if (!this.enabled) {
-      throw new Error('Heap profiler is not enabled.');
-    }
-    const result = profiler.getAllocationProfile();
-    const startTimeNanos = Date.now() * 1000 * 1000;
-    return serializeHeapProfile(result, startTimeNanos, this.intervalBytes);
+/**
+ * Starts heap profiling. If heap profiling has already been started with
```

Comment:
Optional: I think you could as well return an error if an attempt to start already started profiler with different parameters is made as I don't think it should ever be done with the current agent.

Reply:
This is possible with the code as written.
If the user specifies heapIntervalBytes or heapMaxStackDepth in the config pointed to by GCLOUD_PROFILER_CONFIG, then [this](https://github.com/nolanmar511/cloud-profiler-nodejs/blob/193761afd9d42abd41e0b06dc2a175672657390a/ts/src/profiler.ts#L261-L262) line in startProfiler would start the profiler with different parameters... and the profiles could be off.

I can work around this, but only if we decide to never fix #116, because loading the config with require() makes it possible to load the config synchronously.

Reply:
Should this return an error now if start() is called after the profiler is already started?

Reply:
I think that makes sense.
Done.

Reply:
Not done?